sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -30,6 +30,15 @@ trait ObjectOperator extends LogicalPlan {
   /** The serializer that is used to produce the output of this operator. */
   def serializer: Seq[NamedExpression]
 
+  override def output: Seq[Attribute] = serializer.map(_.toAttribute)
+
+  /**
+   * An [[ObjectOperator]] may have one or more deserializers to convert internal rows to objects.
+   * For each deserializer, [[ObjectOperator]] should also provide the attributes that can resolve
+   * this deserializer.
```

Comment:
"An [[ObjectOperator]] must also provide the attributes that are available during the resolution of each deserializer."